### PR TITLE
Update variational inference isoscape generation code to use new model.py constructor

### PR DESCRIPTION
**Pull request by VimWizardVersace**

### CODE REVIEW

File: dnn/variational_inference.ipynb
Comment (erickzul) [inline dnn/variational_inference.ipynb:1]: In line "    MODEL_NAME, vi_model, required_geotiffs, 1000, 1000)", could we explicitly write the kwargs?

**Review by erickzul — CHANGES_REQUESTED** [review]

File: dnn/variational_inference.ipynb [hunk 1]
```diff
@@ -7,7 +7,7 @@
         "colab_type": "text"
```
Comment (benwulfe) [inline dnn/variational_inference.ipynb:7]: <p>this section is unnecessary and probably superfluous for someone not familiar with python.</p><p>you are also importing in the previous code block, so its unclear to someone unfamilar that these are local imports that can be changed.</p><p> </p><p>we should just merge this into the previous section</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tnc-br/ddf-isoscapes/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (benwulfe) [inline dnn/variational_inference.ipynb:7]: <p>combine the next 4 code blocks into a single code block "data preparation" -- unless there are major reasons to execute them independently.</p><p>I also bet there is code here that could be put into ddfcommon, but that is a bigger change to do later.</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tnc-br/ddf-isoscapes/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (benwulfe) [inline dnn/variational_inference.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #1.    def valid_in_all_rasters(lat: float, lon: float, rasters: List[raster.AmazonGeoTiff]) -&gt; bool:
</pre><p>should collapse the data prep code block by default</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tnc-br/ddf-isoscapes/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (benwulfe) [inline dnn/variational_inference.ipynb:7]: <p>include this inside data prep</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tnc-br/ddf-isoscapes/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (benwulfe) [inline dnn/variational_inference.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #1.    from sklearn.metrics import mean_squared_error
</pre><p>please combine the code blocks above into a single click for model definition</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tnc-br/ddf-isoscapes/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (benwulfe) [inline dnn/variational_inference.ipynb:7]: <p>combine all these code blocks into a single block for 'model train'</p><p>maybe separate out 'isoscape generation' from model train.</p><p> </p><p>so:</p><ol><li>parameters</li><li>data prep</li><li>model def</li><li>model train</li><li>save isoscape</li></ol><p> </p><p>5 clicks max if you dont need to change params.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tnc-br/ddf-isoscapes/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Review by benwulfe — CHANGES_REQUESTED** [review]
please combine code blocks and reduce click count.  If code blocks are too big, can refactor out to ddfcommon, but to start, reducing the click count is better especially for a non-expert user of python who may easily miss a code block.

**Comment by VimWizardVersace** [comment]
>please combine code blocks and reduce click count. If code blocks are too big, can refactor out to ddfcommon, but to start, reducing the click count is better especially for a non-expert user of python who may easily miss a code block. 

I found that they don't really look at the number of different sections. They typically would just run runtime -> run all

Otherwise they'd have to manually click to run ~30 sections in the validation pipeline

File: dnn/variational_inference.ipynb (hunk 1 above)
Comment (benwulfe) [inline dnn/variational_inference.ipynb:7]: <pre class="ql-syntax" spellcheck="false">Line #47.        df[name] = df.apply(lambda row: geotiff.value_at(row['long'], row['lat']), axis=1)
</pre><p>could you try using dem() here and see how bad perf is?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tnc-br/ddf-isoscapes/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: dnn/variational_inference.ipynb
Comment (erickzul) [inline dnn/variational_inference.ipynb:1]: In valid_in_all_rasters (could you document the function?)

Comment (erickzul) [reply dnn/variational_inference.ipynb:1]: Also in nudge_invalid_coords

**Review by erickzul — APPROVED** [review]
LGTM pending nits

File: dnn/variational_inference.ipynb
Comment (VimWizardVersace) [reply dnn/variational_inference.ipynb:1]: Ack
